### PR TITLE
DX-116: Properly handle file changes when didChangeWatchedFiles is not a capability

### PR DESCRIPTION
**Pull request by jackyzha0**
- Moves diagnostics calculation to only run once per change
- Adds ability to spawn Chokidar for file changes when `didChangeWatchedFiles` is not advertised as a capability during initialization

### CODE REVIEW

**Review by gburtini — APPROVED** [review]

File: packages/pyright-internal/src/server.ts [hunk 1]
```diff
@@ -81,7 +82,7 @@ export class PyrightServer extends LanguageServerBase {
             openFilesOnly: true,
             useLibraryCodeForTypes: false,
             disableLanguageServices: false,
-            disableOrganizeImports: false,
+            disableOrganizeImports: true,
```
Comment (gburtini) [inline packages/pyright-internal/src/server.ts:85]: is this intentional?

Comment (jackyzha0) [reply packages/pyright-internal/src/server.ts:85]: yeah this was a slightly unrelated change though. we rely on `ruff` for import sorting, not `pyright`

**Comment by masad-frost** [comment]
> Moves diagnostics calculation to only run once per change

Is this related?

**Comment by masad-frost** [comment]
Also pog

**Pull request merged**